hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/exception/S3ErrorTable.java

```diff
@@ -126,6 +126,10 @@ private S3ErrorTable() {
   public static final OS3Exception NO_OVERWRITE = new OS3Exception(
       "Conflict", "Cannot overwrite file with directory", HTTP_CONFLICT);
 
+  public static final OS3Exception METADATA_TOO_LARGE = new OS3Exception(
+      "MetadataTooLarge", "Your metadata headers exceed the maximum allowed " +
+      "metadata size.", HTTP_BAD_REQUEST);
```

Comment:
The msg lacks the originally explicit declaration of MetaData 2KB size. The users usually prefer more clear error msg.

Reply:
I have referred to aws documentation: https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList for the description. Instead of changing it in S3ErrorTable, to make the error message for suitable for this case, I have set the error message in EndpointBase to the originally more explicit message.